### ISSUE TITLE
"No launchable updates found in database" when loading unsupported SDK in development in Expo client

### ISSUE THREAD

**Issue by JChristensen97** [expo/expo#10344]
### Summary of Issue 
Receiving message "No launchable updates found in database" when attempting to run app on iphone 11 simulator.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
```
  Expo CLI 3.27.8 environment info:
    System:
      OS: macOS Mojave 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.18.4 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.14.6 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.8.4 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    IDEs:
      Xcode: 11.2/11B52 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^35.0.0 => 35.0.1
      react: 16.8.3 => 16.8.3
      react-dom: 16.8.3 => 16.8.3
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8
      react-native-web: ^0.11.7 => 0.11.7
      react-navigation: ^4.1.1 => 4.1.1
    npmGlobalPackages:
      expo-cli: 3.27.8
    Expo Workflow: managed
```

### Steps to Reproduce
1. Deploy app via `expo publish` command
2. Attempt to run app locally in IOS iphone 11 simulator via `yarn start` command
3. App does not load and displays message "No launchable updates found in database" on the iphone simulator

### Expected Behavior vs Actual Behavior
**Expected Behavior**: Be able to run the app locally on iphone simulator without receiving the message "No launchable updates found in database".
**Actual Behaviour**: App does not open on simulator and displays message "No launchable updates found in database"

**Comment by brentvatne** [comment]
```
Deploy app via expo publish command
Attempt to run app locally in IOS iphone 11 simulator via yarn start command
App does not load and displays message "No launchable updates found in database" on the iphone simulator
```

i don't understand these steps. if you run expo publish, why would you run `yarn start` after? you've already published the app, you don't need to run `yarn start` to open it, just sign in to your account in expo client and open it from the profile screen

**Comment by brentvatne** [comment]
one thing that can be reproduced:

- run `yarn start` to start the development server.
- run `expo publish` to publish it.
- try to open your app from the development server, it will be unavailable. you need to close the development server and restart it to fix it.

this will be fixed when https://github.com/expo/expo-cli/issues/2322 is enabled by defualt. you can opt in to it now by running `EXPO_USE_DEV_SERVER=1 expo start` and `EXPO_USE_DEV_SERVER=1 expo publish`

but i'm not sure if this is what you are describing, i would need more information.

<hr />

my guess is that you are not signed in to the same account that you published the app from. you need to do this to access the published app in expo client for ios

**Comment by JChristensen97** [comment]
Thanks for your response @brentvatne. 

I understand that I don't need to run `yarn start` to view the published app. I'm actually not trying to view the published app at all, I just included that step as that was the last step I did before I was unable to run the app locally in the simulator.

**Comment by brentvatne** [comment]
ok, so you are running `expo start` and then you try to open that url in the simulator and you see "No launchable updates found in database"? where does `expo publish` come in to play? can you write a new set of repro steps please?

**Comment by JChristensen97** [comment]
Yup, that's correct. I see "No launchable updates found in database" on the simulator after running `expo start`.

`expo publish` might not have anything to do with the issue, I just included it because it was the last command I ran before the app stopped working locally.

**Comment by JChristensen97** [comment]
After clicking on 'try again' to reload the app on the simulator, I get the following message:

```
"React Native version mismatch"

JavaScript version: 0.59.8
Native version: 0.63.2-undefined
```

This message is not displayed every time though. Hope this helps to clarify

**Comment by leslielyj** [comment]
`
"expo": "^35.0.0",
"react": "16.8.3",
"react-native": "https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz",
`

Having the same issue with the exact same versions. Looking for a fix.

**Comment by brentvatne** [comment]
@leslielyj - the latest expo client does not support sdk 35.

**Comment by brentvatne** [comment]
@JChristensen97 - oh sorry, i missed that you are using sdk 35, so this is the same issue as @leslielyj is facing. it's not supported in the latest expo client. see: https://blog.expo.io/expo-sdk-39-is-now-available-4c10aa825e3f - we dropped sdk 35 in this release.

**Comment by brentvatne** [comment]
@esamelson - i renamed this to ""No launchable updates found in database" when loading unsupported SDK in development in Expo client" - we should probably show a more useful error msg

**Comment by JChristensen97** [comment]
Thanks! Updating the SDK to 39.0.0 worked. 

Getting error `Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()"` after updating, but this is a separate issue that I'll need to look into.

**Comment by brentvatne** [comment]
indeed, i'd recommend googling that error

**Comment by leslielyj** [comment]
Just want to clarify that you only need to update Expo's SDK not React Native.

**Comment by brentvatne** [comment]
@leslielyj - please see https://docs.expo.io/workflow/upgrading-expo-sdk-walkthrough/
